tests/unit/select_connection_ioloop_tests.py

```diff
@@ -81,7 +89,54 @@ def start_test(self):
         self.start()
 
     def on_timer(self, val):
-        self.assertEquals(val, self.timer_stack.pop())
+        self.assertEqual(val, self.timer_stack.pop())
+        if not self.timer_stack:
+            self.ioloop.stop()
+
+    def test_normal(self):
+        self.start_test()
+
+    def test_timer_for_suicide(self):
+        self.timer_stack = list()
+        handle_holder = []
+        self.handle = self.ioloop.add_timeout(
+            0.1, partial(self.on_timer_suicide,handle_holder))
+        handle_holder.append(self.handle)
+        self.start()
+
+    def on_timer_suicide(self, handle_holder):
+        self.assertEqual(self.handle, handle_holder.pop())
+        # This removal here should not raise exception by itself nor
+        # in the caller SelectPoller.process_timeouts().
+        self.ioloop.remove_timeout( self.handle )
+        self.ioloop.stop()
+
+    # This patch delays the execution of the handlers so that they get fired
+    # in the same invocation of SelectPoller.process_timeouts().
+    @patch('pika.adapters.select_connection.SelectPoller.is_ready_to_fire',
+           side_effect= lambda x : 0<len(x) and
+                                   x[0][1]['deadline'] < time.time()-0.1 )
+    def test_timers_for_murder(self, mocked_is_ready):
```

Comment:
The use of "suicide" and "murder" in method names is a bit unpleasant. Could you use longer, more descriptive function names that don't involve those ominous words? In unittest, it's preferable to use longer, descriptive method names for test methods. For example: `test_timer_handler_deletes_own_timer` and `test_timer_handler_deletes_another_expired_timer`